File: Segundo Trimestre/6- Ejercicios con Excepciones/python/gatoSimple/GatoSimple.py
"""
GatoSimple.py
Definición de la clase GatoSimple
__author__ : Rafael López | RafaLpeC

"""

from random import randint

"""
Clases del Error de apareamiento

"""
class Error(Exception):
    pass

class ExcepcionApareamientoImposible(Error):
    pass

"""
Clase GatoSimple.
    -Obtener sexo
    -Maullar
    -Ronronear
    -Comer
    -Aparearse con otro gato de su mismo sexo
    -Pelearse con otro gato

"""
class GatoSimple():
    def __init__(self,sexo):
        self.sexo = sexo
    
    def getSexo(self):
        return self.sexo;
    
    def maulla(self):
        print("Miauuuu");
    
    def ronronea(self):
        print("mrrrrrr");
    
    """
    Hacer que el gato coma.
    A los gatos les gusta el pescado, si le damos otra comida
    la rechazará
    """
    def come(self):
        if self.comida == "pescado":
            print("Hmmm, gracias")
        
        else:
            print("Lo siento, yo solo como pescado")
    
    """
    
    
    """
    def apareaCon(self, pareja):
        try:
            if self.sexo == pareja.sexo:
                raise ExcepcionApareamientoImposible

            else:
                if(randint(1,2) == 1):
                    return GatoSimple("hembra")
                else:
                    return GatoSimple("macho")
                
        except ExcepcionApareamientoImposible:
            print("Dos gatos del mismo sexo no pueden criar.")

    def peleaCon(self,contrincante):
        if self.sexo == "hembra":
            print("No me gusta pelear.")
        else:
            if contrincante.sexo == "hembra":
                print("No peleo contra gatitas.")
            else:
                print("Ven aquí que te vas a enterar.")

File: Segundo Trimestre/6- Ejercicios con Excepciones/java/gatoSimple/GatoSimple.java
package gatoSimple;

import GatoSimple.ExcepcionApareamientoImposible;

/**
* GatoSimple.java
* Definición de la clase GatoSimple
* @author Luis José Sánchez
*/


public class GatoSimple {
// atributos /////////////////////////////
  String color, raza, sexo;
  int edad;
  double peso;
// métodos ///////////////////////////////
// constructor
  GatoSimple (String s) {
    this.sexo = s;
  }
// getter
  String getSexo() {
    return this.sexo;
  }
/**
* Hace que el gato maulle
*/
  void maulla() {
    System.out.println("Miauuuu");
  }
/**
* Hace que el gato ronronee
*/
  void ronronea() {
    System.out.println("mrrrrrr");
  }
  /**
  * Hace que el gato coma.
  * A los gatos les gusta el pescado, si le damos otra comida
  * la rechazará.
  *
  * @param comida la comida que se le ofrece al gato
  */
  void come(String comida) {
    if (comida.equals("pescado")) {
      System.out.println("Hmmmm, gracias");
    } else {
      System.out.println("Lo siento, yo solo como pescado");
    }
  }
  /**
  * Pone a pelear dos gatos.
  * Solo se van a pelear dos machos entre sí.
  *
  * @param contrincante es el gato contra el que pelear
  */
  void peleaCon(GatoSimple contrincante) {
    if (this.sexo.equals("hembra")) {
      System.out.println("no me gusta pelear");
    } else {
      if (contrincante.getSexo().equals("hembra")) {
        System.out.println("no peleo contra gatitas");
      } else {
        System.out.println("ven aquí que te vas a enterar");
        }
      }
    }
  
  String apareaCon(GatoSimple pareja) throws ExcepcionApareamientoImposible{
    if(sexo == pareja.sexo) {
      throw new ExcepcionApareamientoImposible("Dos gatos del mismo sexo no pueden criar.");
    }
    else {
      if( (int)(Math.random() * ((2 - 1) + 1)) + 1 == 1) {
        return "hembra";
      }
      else {
        return "macho";
      }
    }
  }
}
